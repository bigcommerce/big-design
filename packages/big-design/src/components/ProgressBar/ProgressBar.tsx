import React from 'react';

import { StyledProgressBar, StyledProgressBarFiller } from './styled';

export interface ProgressBarProps {
  readonly percent?: number;
}

export const ProgressBar: React.FC<ProgressBarProps> = ({ percent }) => {
  return typeof percent === 'number' ? (
    <StyledProgressBar
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={percent}
      role="progressbar"
    >
      <StyledProgressBarFiller percent={percent} />
    </StyledProgressBar>
  ) : (
    <StyledProgressBar role="progressbar">
      <StyledProgressBarFiller />
    </StyledProgressBar>
  );
};
