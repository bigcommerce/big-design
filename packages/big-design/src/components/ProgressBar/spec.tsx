import { render } from '@testing-library/react';
import React from 'react';
import 'jest-styled-components';

import { ProgressBar } from './ProgressBar';

test('render determinant progress bar', () => {
  const { container } = render(<ProgressBar percent={50} />);

  expect(container).toMatchSnapshot();
});

test('render indeterminant progress bar', () => {
  const { container } = render(<ProgressBar />);

  expect(container).toMatchSnapshot();
});
