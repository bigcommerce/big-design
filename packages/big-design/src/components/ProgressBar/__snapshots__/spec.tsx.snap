// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render determinant progress bar 1`] = `
.c0 {
  background-color: #ECEEF5;
  height: 0.25rem;
  overflow: hidden;
  width: 100%;
}

.c1 {
  background-color: #3C64F4;
  height: 100%;
  overflow: hidden;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: width,background-color,height;
  transition-property: width,background-color,height;
  width: 50%;
}

<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="c0"
    role="progressbar"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;

exports[`render indeterminant progress bar 1`] = `
.c0 {
  background-color: #ECEEF5;
  height: 0.25rem;
  overflow: hidden;
  width: 100%;
}

.c1 {
  background-color: #3C64F4;
  height: 100%;
  overflow: hidden;
  -webkit-animation: cdRApH 2s ease-in-out infinite;
  animation: cdRApH 2s ease-in-out infinite;
  position: relative;
  width: 6.25%;
}

<div>
  <div
    class="c0"
    role="progressbar"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;
